### ISSUE TITLE
Bug: wrong username property used when password authentication is not needed

### ISSUE THREAD

**Issue by denisrdasilva** [bbottema/simple-java-mail#429]
Hello! I faced some problem using Simple Java Mail, version 7.5.1, about the prop username. 
When passing the prop username, Jakarta Mail does not recognize it. Debugging the code, I found this: in class org.simplejavamail.api.mailer.config.TransportStrategy line 104:

```java
public String propertyNameUsername() {
	return "mail.smtp.username";
}
```

On the other hand, in jakarta, class jakarta.mail.Service, line 112:

```java
if (protocol != null) {
    if (host == null)
	host = session.getProperty("mail." + protocol + ".host");
    if (user == null)
	user = session.getProperty("mail." + protocol + ".user");
}
```
Simple Mail uses mail.smtp.username, but Jakarta uses mail.smtp.user. Furthemore, I faced an exception about authentication.

To solve this, I found two ways:

```java
public enum TransportStrategy {
....
	public String propertyNameUsername() {
		return "mail.smtp.user";
	}
```
```java
       Properties props = new Properties(); 
       props.put("mail.smtp.user", "any_user");
       MailerImpl mailer = (MailerImpl) MailerBuilder
       ....
       .withProperties(props)
      .buildMailer();
```
So is this problem or is it an error I am causing?

Thanks in advance!

**Comment by bbottema** [comment]
Hmm, it seems you're on to something. For authenticated SMTP servers this is not an issue, as the property is ignored in that case, but for SMTP users that don't require a password, the property is used and it looks like it is the wrong property. I'm still looking into how this is possible, because that property is a core mechanic since the start of the library. It's possibly one of the least tested functions though (passwordless connections).

I'll get back to this.

**Comment by bbottema** [comment]
Your use case is a strange one though; either an SMTP sever requires authentication or it doesn't. If it does, a password is required and then everything is fine as I said. However, if there is no authentication needed by the server, not even the user name is required. I suppose it depends on the server implementation how it deals with missing username, but the test server [Wiser](https://github.com/voodoodyne/subethasmtp/blob/master/Wiser.md), which I use in my junit tests, doesn't have any issue no matter the configuration.

**Comment by bbottema** [comment]
Release fix in 7.5.2. Give it a spin!

**Comment by denisrdasilva** [comment]
Thank you and happy new year!
    Em domingo, 25 de dezembro de 2022 08:43:01 BRT, Benny Bottema ***@***.***> escreveu:  
 
 


Release fix in 7.5.2. Give it a spin!
